File: api/internal/domain/models/payment.go
package models

import "time"

type Payment struct {
	ID            string              `json:"id"`
	UserID        *string             `json:"userId"` // Nullable for unclaimed payments
	ExternalName  string              `json:"externalName"`
	LeagueID      string              `json:"leagueId"`
	AmountCents   int                 `json:"amountCents"`
	Method        PaymentMethod       `json:"method"`
	ReceivedAt    time.Time           `json:"receivedAt"`
	RecordedBy    string              `json:"recordedBy"`
	Reference     *string             `json:"reference"`
	Allocations   []PaymentAllocation `json:"allocations"`
}

func (p Payment) UnallocatedAmount() int {
	allocated := 0
	for _, a := range p.Allocations {
		allocated += a.AmountInCents
	}
	return p.AmountCents - allocated
}

func CreatePayment(userID *string, externalName string, leagueID string, amount int, method PaymentMethod, reference *string) *Payment {
	return &Payment{
		UserID:        userID,
		ExternalName:  externalName,
		LeagueID:      leagueID,
		AmountCents:   amount,
		Method:        method,
		ReceivedAt:    time.Now(),
		Reference:     reference,
		Allocations:   make([]PaymentAllocation, 0),
	}
}

File: bouncy-web/src/lib/models/payment.ts
import type { PaymentAllocation } from './payment-allocation';

export type PaymentMethod = 'venmo' | 'cash';

export interface Payment {
  id: string;
  userId?: string | null;
  externalName: string;
  leagueId: string;
  amountInCents: number;
  method: PaymentMethod;
  receivedAt: string | Date;
  recordedBy: string;
  reference?: string | null;
  allocations: PaymentAllocation[];
}
